### PR TITLE
Task/wi 33

### PR DESCRIPTION
**Pull request by taoteg**
## Overview

Update to the verbiage and technologies described in the Portal Architecture Guide available in the Core-CMS.

## Related

- [WI-33](https://jira.tacc.utexas.edu/browse/WI-33)

## Changes

- Updated the template guide page of static content with current tech.
- Replace all references to Angular with React.
- Replace all references to Agave with TAPIS.
- Cleanup the general language and make it more readable.

## Testing

- TL;DR: Review the text edits in the github commit. Else...

1. Pull branch.
2. Update your `settings_local.py` to include:
```
CMS_TEMPLATES = (
    ...
    ('guides/portal_technology.html', 'Guide: Portal Technology Stack'),
)
```
4. `make build`
5. `make start`
6. In the CMS Admin, create a new CMS page using the `guides/portal_technology.html` as the template and publish it.
7. Verify the page content looks correct.

## UI

- Text edits.

## Notes

- None.

### CODE REVIEW

File: taccsite_cms/templates/guides/portal_technology.html [hunk 1]
```diff
@@ -5,36 +5,40 @@ <h1>Portal Technology Stack</h1>
...
+  <p>The TACC CEP, short for Core Experience Portal, is the core engine for all TACC portal projects. The primary objective for the Core Experience Portal is to establish and grow a codebase that can be used as a common baseline for all future portal projects undertaken at TACC. By cultivating this common codebase for all of our portal projects, we maintain better alignment between the core capabilities and technologies supported across all of the TACC portals. While some projects do have unique requirements, CEP provides an "out-of-the-box" framework for rapidly deploying a ready-to-use project infrastructure that includes all the common TACC portal capabilities pre-configured and in compliance with TACC's current best practices.</p>
 
-  <h3>Common Portal Capabilities:</h3>
+  <h3>Common Portal Capabilities</h3>
```
Comment (taoteg) [inline taccsite_cms/templates/guides/portal_technology.html:11]: The whole Common Portal Capabilities list feels a little off. Later I was more diligent about distinguishing between types of systems, etc. I am not sure this section is consistent with the rest of my edits. Thoughts?

File: taccsite_cms/templates/guides/portal_technology.html [hunk 2]
```diff
@@ -45,115 +49,118 @@ <h2>High-level Architecture</h2>
...
+  <p>Applications are executable code available for invocation through the TAPIS Jobs service on a specific execution system. If a single code needs to be run on multiple systems, each combination and system needs to be defined as an app. Code that can be forked at the command line or submitted to a batch scheduler can be registered as a TAPIS app and run through the Jobs service. The user sends a request by populating the application inputs and job details in the portal. Applications are packaged together using three files:</p>
 
   <dl class="s-inline-dl  u-nested-text-content">
-    <dt>App.json</dt><dd>JSON file with application definitions and parameters such as (name, version, label).</dd>
-    <dt>Wrapper.sh</dt><dd>wrapper file, shell script which executes the job on HPC/Docker.</dd>
-    <dt>Test.sh</dt><dd>runs a test before the job is launched and cleans up the files.</dd>
+    <dt>App.json</dt><dd>A JSON file with application definitions and input parameters (i.e. - name, version, label).</dd>
+    <dt>Wrapper.sh</dt><dd>A shell script which executes the job on the targeted resource (i.e. - an HPC system or virtualized container).</dd>
```
Comment (wesleyboar) [inline taccsite_cms/templates/guides/portal_technology.html:119]: I haven't seen "i.e. - " before...

```suggestion
    <dt>App.json</dt><dd>A JSON file with application definitions and input parameters (i.e. name, version, label).</dd>
    <dt>Wrapper.sh</dt><dd>A shell script which executes the job on the targeted resource (i.e. an HPC system or virtualized container).</dd>
```

File: taccsite_cms/templates/guides/portal_technology.html [hunk 3]
```diff
@@ -45,115 +49,118 @@ <h2>High-level Architecture</h2>
...
 
   <p>The Core portal utilizes a wide variety of technologies developed by multiple technology vendors. Below is a list of the primary libraries, frameworks and APIs leveraged in the core portal tech stack.</p>
 
   <dl class="s-inline-dl">
+    <dt>React</dt>
+      <dd>The JavaScript framework providing all client-side functionality. This is the broswer-based client the user interacts with when accessing the portal.</dd>
+
     <dt>NginX</dt>
-      <dd>When user sends http request to call the CEP URL, it goes to nginx webserver, a proxy, intermediary for requests from clients seeking resources from other&nbsp;servers. Nginx carry the request to the Web Server Gateway.</dd>
+      <dd>The nginx webserver is as a proxy for all portal users, acting as intermediary for requests from clients seeking resources from the server. Nginx curries the request to the appropriate backend server.</dd>
```
Comment (wesleyboar) [inline taccsite_cms/templates/guides/portal_technology.html:140]: Do you mean "Nginx carries the request" or "Nginx proxies the request", instead of "… curries …"?

**Review by wesleyboar — CHANGES_REQUESTED** [review]
Nit picking.

File: taccsite_cms/templates/guides/portal_technology.html (hunk 2 above)
Comment (taoteg) [reply taccsite_cms/templates/guides/portal_technology.html:119]: ahh, the dash.... good catch!

File: taccsite_cms/templates/guides/portal_technology.html (hunk 3 above)
Comment (taoteg) [reply taccsite_cms/templates/guides/portal_technology.html:140]: Was using the generic courier definition (but is confusing with there being a concept of code currying) so I'll make it `routes` as that is more correct.

File: taccsite_cms/templates/guides/portal_technology.html [hunk 4]
```diff
@@ -5,36 +5,40 @@ <h1>Portal Technology Stack</h1>
...
-  <h3>CEP Major Components:</h3>
+  <h3>CEP Major Components</h3>
 
   <ul>
-    <li>General Content Management</li>
+    <li>Content Management</li>
     <li>My Account</li>
     <li>My Dashboard</li>
     <li>Data Files (Data Depot)</li>
     <li>Applications (workspace)</li>
```
[the remark this answers is not in the document]
Comment (taoteg) [reply taccsite_cms/templates/guides/portal_technology.html:35]: Dunno, it is vestigial from the old text. I almost removed it myself but wasn't sure if tere was a purpose. Going to cull it now!

File: taccsite_cms/templates/guides/portal_technology.html [hunk 5]
```diff
@@ -45,115 +49,118 @@ <h2>High-level Architecture</h2>
...
+  <p>The dashboard presents users with a unified interface to enable their research. The <em>Data Files</em> view is where users can manage their data and shared data resources. The <em>Applications</em> view allows users to select and execute various applications (both remote execution and interactive jobs). The <em>Allocations</em> view allows users to view and request available allocation resources for their projects as well as monitor who has access to said resources. The <em>History</em> view logs all jobs the user has executed (or is currently executing) with options to view the job details and output location. The <em>System Status</em> view displays detailed operational status information about the available HPC systems, including queue states, jobs currently running/waiting in queue, and the number of idle nodes on the resource.</p>
```
[the remark this answers is not in the document]
Comment (taoteg) [reply taccsite_cms/templates/guides/portal_technology.html:99]: Going dd

**Comment by taoteg** [comment]
@wesleyboar if you get a second, can you review it once more for correctness?

File: taccsite_cms/templates/guides/portal_technology.html (hunk 3 above)
Comment (wesleyboar) [reply taccsite_cms/templates/guides/portal_technology.html]: Ah, okay. That was new term and concept for me.

**Review by wesleyboar — COMMENTED** [review]
A `<dl>` may not be nested directly within a `<dl>`.

- #732

Also, I tweaked some text for accuracy, like in "… Major Components" list.

**Review by wesleyboar — APPROVED** [review]

File: taccsite_cms/static/site_cms/css/src/_migrations/v3-10_v3-11/site.template.guide.css
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_migrations/v3-10_v3-11/site.template.guide.css:1]: This rare change to a minified archived file is tested and documented in #732.

File: taccsite_cms/templates/guides/portal_technology.html (hunk 1 above)
Comment (wesleyboar) [reply taccsite_cms/templates/guides/portal_technology.html:11]: Well… mentioning Elastic Search is unique. No other bullet in this list mentions a software. And all other proper nouns are suggestive of a component using non-software terms.

Comment (taoteg) [reply taccsite_cms/templates/guides/portal_technology.html:11]: I am a little confused on the state of our portal search engine - are we still using ES or have we transitioned to Google for all things search now?

Comment (wesleyboar) [reply taccsite_cms/templates/guides/portal_technology.html:11]: Suggestions, drop Elastic Search here, or move it to "Environment" and reword this to just be—

> **Search**

—or—

> An **Integrated Search**

Comment (wesleyboar) [reply taccsite_cms/templates/guides/portal_technology.html:11]: We have not transitioned to Google for CMS search. That transition comes piecemeal. Elastic Search is still used for "behind-the-login" search.

Comment (wesleyboar) [reply taccsite_cms/templates/guides/portal_technology.html:11]: It's fine as is. Sorry for any brain belaboring I caused. I haven't heard @jarosenb mention abandoning Elastic Search for anything other than CMS.
